crates/wasi-provider/src/wasi_runtime.rs

```diff
@@ -355,18 +350,27 @@ impl WasiRuntime {
     }
 }
 
-fn send(mut sender: Sender<Status>, name: String, status: Status, cx: &mut Context<'_>) {
+fn send(sender: &Sender<Status>, name: &str, status: Status) {
+    use tokio::sync::mpsc::error::TrySendError;
+    debug!("{} sending status: {:?}", name, &status);
+    // match sender.blocking_send(status) {
+    //     Err(e) => warn!("{} error sending wasi status: {:?}", name, e),
+    //     Ok(_) => debug!("{} send completed.", name),
+    // }
     loop {
-        if let Poll::Ready(r) = sender.poll_ready(cx) {
-            if r.is_ok() {
-                sender.try_send(status).expect("Possible deadlock, exiting");
+        match sender.try_send(status.clone()) {
+            Ok(()) => {
+                debug!("{} send completed.", name);
+                return;
+            }
+            Err(TrySendError::Full(_)) => {
+                debug!("{} status channel full.", name);
+            }
+            Err(TrySendError::Closed(_)) => {
+                warn!("{} status channel closed.", name);
                 return;
             }
-            trace!("Receiver for status showing as closed: {:?}", r);
         }
-        trace!(
-            "Channel for container {} not ready for send. Attempting again",
-            name
-        );
+        std::thread::sleep(std::time::Duration::from_millis(100));
     }
 }
```

Comment:
Are you aware that we added a [`blocking_send`](https://docs.rs/tokio/1/tokio/sync/mpsc/struct.Sender.html#method.blocking_send) function?

Reply:
Yes! We actually need to change this back. In our flailing around to fix the deadlock problem, we wanted to eliminate anything that was a new Tokio 1.0 API as a sanity check